grid-proxy/pkg/client/grid_client_test.go

```diff
@@ -28,6 +30,67 @@ var (
 	NodeStatusExample = MarshalNodeStatus([]byte(NodeStatusExampleStr))
 )
 
+func nodesFilterValues() (types.NodeFilter, types.Limit, string) {
+	Up := "up"
+	Egypt := "Egypt"
+	Mansoura := "Mansoura"
+	Freefarm := "Freefarm"
+	trueVal := true
+	falseVal := false
+	ints := []uint64{0, 1, 2, 3, 4, 5, 6}
+	f := types.NodeFilter{
+		Status:       &Up,
+		FreeMRU:      &ints[1],
+		FreeHRU:      &ints[2],
+		FreeSRU:      &ints[3],
+		Country:      &Egypt,
+		City:         &Mansoura,
+		FarmName:     &Freefarm,
+		FarmIDs:      []uint64{1, 2},
+		FreeIPs:      &ints[4],
+		IPv4:         &trueVal,
+		IPv6:         &falseVal,
+		Domain:       &trueVal,
+		Rentable:     &falseVal,
+		RentedBy:     &ints[5],
+		AvailableFor: &ints[6],
+	}
+	l := types.Limit{
+		Page: 12,
+		Size: 13,
+	}
+	return f, l, "status=up&free_mru=1&free_hru=2&free_sru=3&country=Egypt&city=Mansoura&farm_name=Freefarm&farm_ids=1&farm_ids=2&free_ips=4&ipv4=true&ipv6=false&domain=true&rentable=false&rented_by=5&available_for=6&page=12&size=13&ret_count=false&randomize=false"
```

Comment:
shouldn't we encode instead of writing the string?

Reply:
from what I gather, he hard-codes the expected values here from the defined f and l
then in the tests, can assert the equality of this string value to the ones from decoding both (f, l)